core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -682,7 +682,7 @@ protected Map<String, String> summary() {
   }
 
   @Override
-  public List<ManifestFile> apply(TableMetadata base) {
+  public List<ManifestFile> apply(TableMetadata base, Long parentSnapshotId) {
     Snapshot current = base.currentSnapshot();
```

Comment:
Can we make use of `parentSnapshotId` here ? Anyway we are defaulting target branch to MAIN, so should not be an issue.

Reply:
Yeah, done